### PR TITLE
Replace nivo with recharts for pie chart in envoy triage

### PR DESCRIPTION
**Pull request by smonero**
### Description

The goal here is to replace the nivo pie chart with a recharts pie chart.

before
Screen Shot 2022-03-03 at 10 12 52 AM

after
Screen Shot 2022-03-03 at 10 12 35 AM

### Testing Performed

staging

### GitHub Issue

Fixes #

### CODE REVIEW

File: frontend/workflows/envoy/src/remote-triage/dashboard.tsx [hunk 1]
```diff
@@ -47,20 +47,22 @@ interface FeaturedSummaryProps {
 const FeaturedSummary = ({ summary }: { summary: FeaturedSummaryProps }) => {
   const values = summary?.data?.map(d => d.value);
   const total = values.reduce((t, value) => t + value);
+  // We transform the data from the summary format to a format used for Recharts Library
+  const pieChartData = summary?.data?.map(d => {
```
Comment (jdslaugh) [inline frontend/workflows/envoy/src/remote-triage/dashboard.tsx:51]: Is there a reason we're casting this to the same thing? It looks like the summary format is identical so couldn't we just do `summary?.data || []` as it was previously?

File: frontend/workflows/envoy/src/remote-triage/dashboard.tsx [hunk 2]
```diff
@@ -47,20 +47,22 @@ interface FeaturedSummaryProps {
 const FeaturedSummary = ({ summary }: { summary: FeaturedSummaryProps }) => {
   const values = summary?.data?.map(d => d.value);
   const total = values.reduce((t, value) => t + value);
```
Comment (jdslaugh) [inline frontend/workflows/envoy/src/remote-triage/dashboard.tsx:49]: I know this isn't part of the PR, but I'd adjust this a bit. We're iterating over values to pull them out to then iterate over them for a total.

```suggestion
  const total = (summary?.data || []).reduce((t, { value = 0 }) => t + value, 0);
```

File: frontend/workflows/envoy/src/remote-triage/dashboard.tsx [hunk 3]
```diff
@@ -44,23 +44,31 @@ interface FeaturedSummaryProps {
...
+            <Pie
+              data={summary?.data || []}
+              dataKey="value"
+              innerRadius={PIECHART_INNER_RADIUS}
+              outerRadius={PIECHART_OUTER_RADIUS}
+              legendType="none"
+              animationDuration={PIECHART_ANIMATION_DURATION_MS}
+            >
+              {summary?.data?.map(d => (
+                <Cell key={`cell-${d.id}-${d.value}-${d.color}`} fill={d.color} />
```
Comment (danielhochman) [inline frontend/workflows/envoy/src/remote-triage/dashboard.tsx:68]: i wouldn't include the value with the key, or probably the color. just the id.

**Review by danielhochman — APPROVED** [review]

**Review by jdslaugh — APPROVED** [review]
Thanks!

**Pull request merged**